prime-router/docs/proposals/0013-batch-retry.md

```diff
@@ -0,0 +1,71 @@
+# Batch Process Update / Retry
+The batching process could use more robust retry functionality and there is an opportunity for additional improvements
+to the functionality as a whole to address scalability and performance concerns.
+
+Associated Ticket - #2902
+
+### Current Batch Functionality  
+Each incoming message puts an entry into the 'batch' queue per receiver that message is going to, using the hidden
+'next action at' part of the queue and selecting the next 'batch bucket' for the associated receiver.
+
+When that time comes all queue messages for the receiver in question are added to the queue and picked up by individual 
+instances of the batch function (one per message on the queue). 
+
+Each of these functions then race to select and lock 
+all records from the Task table that are for the receiver in question. The function instances that fail to get any
+records die.
+
+If a batch function fails to succeed it will be lost in the ether and require manual resetting to pick up and batch
+included records.
+
+### Proposed New Functionality
+We can improve this situation and add retry/scalability by adding a cron job and altering the way we use the batch
+queue. Instead of adding each message-per-receiver-per-bucket to the queue at ingestion a cron job could run at
+the minimum granularity for batching (1 minute), determine which receivers need to be batched at that time, and put a 
+message on the queue to be handled immediately by the batchFunction. This is a minimal-change approach that leaves the
```

Comment:
```suggestion
We can improve this situation and add retry/scalability by adding a cron job and altering the way we use the batch
queue. Instead of adding each message-per-receiver-per-bucket to the queue at ingestion a cron job could run at
the minimum granularity for batching (1 minute), determine which receivers need to be batched at that time, and put a 
message on the queue to be handled immediately by the batchFunction. 

The Batch function will be changed to grab all Tasks whose next_action_at is in the past, thereby creating a natural retry mechanism.  (Currently Batch requires an exact match of dates). 

This is a minimal-change approach that leaves the
```